src/components/ToDo/index.tsx

```diff
@@ -0,0 +1,210 @@
+}) => {
+  const { title, userId, id, completed } = todo;
+  const [updateTodo, setUpdateTodo] = useState<Todo | null>(null);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const titleField = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    titleField.current?.focus();
+  });
+
+  const handleSubmit = async () => {
+    setIsSubmitting(true);
+    onError('');
+    onIdTodo(id);
+
+    if (!updateTodo) {
+      setIsSubmitting(false);
+
+      return;
+    }
+
+    if (!updateTodo.title.trim()) {
+      onIdTodo(updateTodo.id);
+      onDelete(updateTodo.id)
+        .then(() => {
+          setUpdateTodo(null);
+          onIdTodo(updateTodo.id);
+        })
+        .finally(() => {
+          onLoading(false);
+          onIdTodo(0);
+        });
+    } else {
+      onIdTodo(0);
+      const todoActual = todos.find(item => item.id === updateTodo.id);
+
+      if (todoActual && updateTodo.title === todoActual.title) {
+        setUpdateTodo(null);
+        setIsSubmitting(false);
+
+        return;
+      }
+
+      onIdTodo(updateTodo.id);
+      onLoading(true);
+      onUpdate({ ...updateTodo, title: updateTodo.title.trim() })
+        .then(() => {
+          setUpdateTodo(null);
+          onIdTodo(updateTodo.id);
+        })
+        .finally(() => {
+          onLoading(false);
+          onIdTodo(0);
+        });
+    }
+
+    setTimeout(() => {
+      setIsSubmitting(false);
+    }, 1000);
+  };
+
+  const handleChecked = (post: Todo) => {
+    onLoading(true);
+    onIdTodo(post.id);
+    const updateCompletedTodo = {
+      ...todo,
+      completed: !post.completed,
+    };
+
+    onUpdate(updateCompletedTodo)
+      .then(() => setUpdateTodo(null))
+      .finally(() => {
+        onLoading(false);
+        onIdTodo(0);
+      });
+  };
+
+  const handleKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      setUpdateTodo(null);
+
+      return;
+    }
+
+    if (event.key === 'Enter') {
+      if (!isSubmitting) {
+        handleSubmit();
+      }
+    }
+  };
+
+  const handleChangeTitle = (event: ChangeEvent<HTMLInputElement>) => {
+    if (updateTodo) {
+      setUpdateTodo(currentTodo => {
+        if (!currentTodo) {
+          return null;
+        }
+
+        return {
+          ...currentTodo,
+          title: event.target.value,
+        };
+      });
+    }
+  };
+
+  const handleBlur = () => {
+    if (!isSubmitting) {
+      handleSubmit();
+    }
+
+    return;
+  };
+
+  return (
+    <div data-cy="Todo" className={`todo ${completed && 'completed'}`} key={id}>
+      {/* eslint-disable jsx-a11y/label-has-associated-control  */}
+      <label className="todo__status-label" htmlFor={'' + id}>
+        <input
+          id={'' + id}
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={completed}
+          onClick={() => handleChecked({ title, id, userId, completed })}
```

Comment:
```suggestion
          onClick={handleChecked}
```
you can take this parameters from props inside function